src/interlocking_component/route_controller.py

```diff
@@ -0,0 +1,16 @@
+class RouteController(Component):
+    """This class coordinates the route of a train.
+    It calls the router to find a route for a train.
+    It makes sure, that the Interlocking sets fahrstrassen along those routes.
+    """
+
+    def next_tick(self, tick: int):
+        """This may be called to prosses a new simulation tick.
+
+        Args:
+            tick (int): The current tick of the simulation.
+
+        Raises:
+            NotImplementedError: This has not been Implemented yet.
+        """
```

Comment:
I think this does not match the sphinx format. If you use autoDocstring you can set the docstring format to sphinx in the extension settings.